janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/builder/CQLStoreFeaturesBuilder.java

```diff
@@ -61,7 +61,7 @@ public CQLStoreFeaturesWrapper build(final CqlSession session, final Configurati
         fb.keyConsistent((onlyUseLocalConsistency ? local : global), local);
         fb.locking(useExternalLocking);
         fb.optimisticLocking(true);
-        fb.multiQuery(false);
+        fb.multiQuery(true);
```

Comment:
Maybe I missed something, but I remember `IN` query in Cassandra was an anti-pattern?

Reply:
Yes, ‘IN’ query is anti-pattern in Cassandra. That’s why we don’t use it for mutliQuery implementation. Instead we are reusing the same getSlice method which we use for normal (non-multiQuery) slices access and then simply await for all responses. I.e. we execute a separate CQL query for each of the provided keys during multiQuery evaluation in CQL backend implementation.

Reply:
The reason why it’s now possible is because getSlice function is changed to be non-blocking and returning future result immediately.
Thus, we can send queries for each key separately and then await for all results to be processed.

Reply:
I see, so we don't need to rely on the backend threading pool anymore to realize multi-query any more. Brilliant!

Reply:
Yes, we are not using a thread pool to send CQL requests in parallel anymore :) Instead it’s now async approach.